### PR TITLE
Moved plugin-ssh-key from plugins to core modules of ws-agent

### LINKED ISSUE #4483: Make plugin-ssh-key as core artifact to plugin
plugin-ssh-key contains some basic functionality that is reused by other core services like Jgit implementation of core git service. We would like to convert plugin-ssh-key to core part.

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
This pr move
    plugins/plugin-ssh-key/che-plugin-ssh-key-ide    -> wsagent/che-core-ssh-key-ide
    plugins/plugin-ssh-key/che-plugin-ssh-key-server -> wsagent/che-core-ssh-key-server

groupid and artifactid of artifacts remain the same.

In next releases groupid and artifactid of artifacts will be changed from 
```
    <groupId>org.eclipse.che.plugin</groupId>
    <artifactId>che-plugin-ssh-key-server</artifactId>
```
to
```
    <groupId>org.eclipse.che.core</groupId>
    <artifactId>che-core-ssh-key-server</artifactId>
```
and
```
     <groupId>org.eclipse.che.plugin</groupId>
    <artifactId>che-plugin-ssh-key-ide</artifactId>

```
to
```
     <groupId>org.eclipse.che.core</groupId>
    <artifactId>che-core-ssh-key-ide</artifactId>
```
### What issues does this PR fix or reference?
Fix https://github.com/eclipse/che/issues/4483

#### Changelog
Moved `plugin-ssh-key` from `plugins` to `wsagent` packages. 

#### Release Notes
In this release, we have moved `plugin-ssh-key` into new packages:
    plugins/plugin-ssh-key/che-plugin-ssh-key-ide    -> wsagent/che-core-ssh-key-ide
    plugins/plugin-ssh-key/che-plugin-ssh-key-server -> wsagent/che-core-ssh-key-server

In this version `groupid` and `artifactid` remains the same, however, in the next release `groupid` and `artifactid` for these artifacts will be changed from plugin to core:

```
    <groupId>org.eclipse.che.plugin</groupId>
    <artifactId>che-plugin-ssh-key-server</artifactId>
```

to

```
    <groupId>org.eclipse.che.core</groupId>
    <artifactId>che-core-ssh-key-server</artifactId>
```

Also:

```
    <groupId>org.eclipse.che.plugin</groupId>
    <artifactId>che-plugin-ssh-key-ide</artifactId>

```

will move to

```
    <groupId>org.eclipse.che.core</groupId>
    <artifactId>che-core-ssh-key-ide</artifactId>
```

### CODE REVIEW

**Review by vparfonov — APPROVED** [review]

File: wsagent/che-core-ssh-key-server/pom.xml [hunk 1]
```diff
@@ -11,16 +11,18 @@
       Codenvy, S.A. - initial API and implementation
 
 -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (sleshchenko) [inline wsagent/che-core-ssh-key-server/pom.xml]: please run `mvn sortpom:sort`

Comment (skabashnyuk) [reply wsagent/che-core-ssh-key-server/pom.xml]: done

**Review by riuvshin — APPROVED** [review]

**Comment by JamesDrummond** [comment]
@skabashnyuk Try to keep the changlelog to a single line of a list of lines. Moved your information up to description.

**Review by JamesDrummond — APPROVED** [review]

**Comment by bmicklea** [comment]
Moved the informative content into the release notes section. I *think* this change impacts the community and should be documented in release notes at least.

**Comment by bmicklea** [comment]
@skabashnyuk how large is the impact on (a) users of Che and (b) people building custom assemblies on Che?

**Comment by skabashnyuk** [comment]
this pr has no effect on building custom assemblies on Che. Because maven groupId and artifactId wasn't changed. only location in source tree.  But we are going to change it in next releases to fit it logical place.

**Review by bmicklea — APPROVED** [review]

**Comment by skabashnyuk** [comment]
ci-build

**Pull request merged**